libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -455,6 +463,26 @@ AP_GPS_UBLOX::_request_next_config(void)
 #endif
         break;
 
+    case STEP_F9: {
+        if (_hardware_generation == UBLOX_F9) {
+            uint8_t cfg_count = populate_F9_gnss();
+            // special handling of F9 config
+            if (cfg_count > 0) {
+                CFG_Debug("Sending F9 settings, GNSS=%u", params.gnss_mode);
+                if (!_configure_list_valset(config_GNSS, cfg_count, UBX_VALSET_LAYER_RAM | UBX_VALSET_LAYER_BBR)) {
+                    _next_message--;
+                    break;
+                }
+                // the call will reset the GNSS subsystem so spec says to wait a further 0.5s before sending more messages
+                hal.scheduler->delay(500);
```

Comment:
this is unsafe, has to be state machine, this runs in main thread

Reply:
Fixed